app/controllers/guide_progress_controller.rb

```diff
@@ -1,7 +1,18 @@
 class GuideProgressController < ApplicationController
+  before_action :check_not_in_exam!
```

Comment:
Maybe we should specify that this action is only for destroy, to make it more action revealing for the future.

Reply:
yeah, that's fair